### DISCUSSION TITLE
Communication Between Agents

### DISCUSSION THREAD

**Discussion by taskon** [metadriverse/metadrive#512] [Q&A] [answered]
Hello, does the environment support explicit communication between agents, for example to use algorithms like CommNet or IC3Net, or do I need to find a workaround to add communication? Thanks!

**Comment by QuanyiLi** [comment]
@pengzhenghao Could you help with this?

**Comment by pengzhenghao** [comment, marked as answer]
The environment does not support cross agent communication. But it's easy to wrap the environment with:

1) extra action space for each agent rather than the 2D steering-acceleration action.
2) a dispatching mechanism send the message from each agent to the correct target agents.
3) add the communication to the observation space so the return from env.step() will include the communication from other agents.

**Reply by taskon** [reply]
Perfect! thank you so much for replying.
